pkg/network/ovn_kubernetes.go

```diff
@@ -958,53 +977,69 @@ func getMasterAddresses(kubeClient crclient.Client, controlPlaneReplicaCount int
 		for i := 0; i < controlPlaneReplicaCount; i++ {
 			ovnMasterAddresses = append(ovnMasterAddresses, fmt.Sprintf("ovnkube-master-%d.ovnkube-master-internal.%s.svc.cluster.local", i, os.Getenv("HOSTED_CLUSTER_NAMESPACE")))
 		}
-	} else {
-		err := wait.PollImmediate(OVN_MASTER_DISCOVERY_POLL*time.Second, time.Duration(OVN_MASTER_DISCOVERY_TIMEOUT)*time.Second, func() (bool, error) {
-			matchingLabels := &crclient.MatchingLabels{"node-role.kubernetes.io/master": ""}
-			if err := kubeClient.List(context.TODO(), masterNodeList, matchingLabels); err != nil {
-				return false, err
-			}
-			if len(masterNodeList.Items) != 0 && controlPlaneReplicaCount == len(masterNodeList.Items) {
-				return true, nil
-			}
+		return ovnMasterAddresses, nil
+	}
 
-			heartBeat++
-			if heartBeat%3 == 0 {
-				klog.V(2).Infof("Waiting to complete OVN bootstrap: found (%d) master nodes out of (%d) expected: timing out in %d seconds",
-					len(masterNodeList.Items), controlPlaneReplicaCount, OVN_MASTER_DISCOVERY_TIMEOUT-OVN_MASTER_DISCOVERY_POLL*heartBeat)
-			}
-			return false, nil
-		})
-		if wait.ErrWaitTimeout == err {
-			klog.Warningf("Timeout exceeded while bootstraping OVN, expected amount of control plane nodes (%v) do not match found (%v): %s, continuing deployment with found replicas", controlPlaneReplicaCount, len(masterNodeList.Items))
-			// On certain types of cluster this condition will never be met (assisted installer, for example)
-			// As to not hold the reconciliation loop for too long on such clusters: dynamically modify the timeout
-			// to a shorter and shorter value. Never reach 0 however as that will result in a `PollInfinity`.
-			// Right now we'll do:
-			// - First reconciliation 250 second timeout
-			// - Second reconciliation 130 second timeout
-			// - >= Third reconciliation 10 second timeout
-			if OVN_MASTER_DISCOVERY_TIMEOUT-OVN_MASTER_DISCOVERY_BACKOFF > 0 {
-				OVN_MASTER_DISCOVERY_TIMEOUT = OVN_MASTER_DISCOVERY_TIMEOUT - OVN_MASTER_DISCOVERY_BACKOFF
-			}
-		} else if err != nil {
-			return nil, fmt.Errorf("Unable to bootstrap OVN, err: %v", err)
+	// Not Hypershift... find all master nodes by label
+	err := wait.PollImmediate(OVN_MASTER_DISCOVERY_POLL*time.Second, time.Duration(timeout)*time.Second, func() (bool, error) {
+		matchingLabels := &crclient.MatchingLabels{"node-role.kubernetes.io/master": ""}
+		if err := kubeClient.List(context.TODO(), masterNodeList, matchingLabels); err != nil {
```

Comment:
Create a context with timeout from func parameter `timeout`?

Reply:
Done. Per below about backoff though, I don't think backoff needs to be passed in? It's constant and the tests don't really care right now.